### ISSUE TITLE
[🐞] qwik-city: vite base doesn't work in dev mode

### ISSUE THREAD

**Issue by wmertens** [QwikDev/qwik#5984]
### Which component is affected?

Qwik City (routing)

### Describe the bug

When you set `config.base` of vite, the build correctly uses it but dev mode breaks, getting module files without the base path.

### Reproduction

https://stackblitz.com/edit/qwik-starter-wvwual?file=vite.config.ts

### Steps to reproduce

try clicking Home

### System Info

```shell
stackblitz, 1.5.1
```

**Comment by wmertens** [comment]
Workaround: In your vite config, use

```ts
export default defineConfig(({command, mode}) => {
	return {
		base: command === 'build' ? '/your-base/' : undefined,
// ...
```

**Comment by JerryWu1234** [comment]
let me pick this up.,

**Comment by JerryWu1234** [comment]
![image](https://github.com/BuilderIO/qwik/assets/27261522/240e74f1-8cf6-437a-b08c-875d24913c42)
you say this bug ?

**Comment by wmertens** [comment]
@JerryWu1234 yes it should use the import.meta.env.BASE_URL in front of it

**Comment by JuanmaVargas** [comment]
In production, is there any workaround to fix this? I'm trying to create a multi-zone in Next.js for a Qwik application, and for that, I need the vite.base set in the configuration to work for all the assets

**Comment by wmertens** [comment]
@JuanmaVargas do you mean it's not working in production? 

I have the fix in #6326 but it's part of a rewrite. I hope to land it soon.

**Comment by JohnPremKumar** [comment]
Meantime if anyone looking for workaround set different path production and development:

```js
export default defineConfig(() => {
  return {
    base: process.env.NODE_ENV === 'production' ? `/production-path/` : '',
    plugins: [
      qwikCity(),
      qwikVite(),
      tsconfigPaths(),
      vanillaExtractPlugin(),
    ],
  };
});
```

**Comment by wmertens** [comment]
I believe this to be fixed in 1.7.1. There are still some problems with windows though, see #6676
